File: src/main/java/com/vc/medium/L858.scala
package com.vc.medium

object L858 {
  def mirrorReflection(p: Int, q: Int): Int = {
    var pVar = p
    var qVar = q
    while(pVar % 2 == 0 && qVar % 2 == 0) {
      pVar /= 2
      qVar /= 2
    }
    if(pVar % 2 == 0) 2
    else if(qVar % 2 == 0) 0
    else 1
  }
}

File: src/main/java/com/vc/medium/L858.java
package com.vc.medium;

class L858 {
    public int mirrorReflection(int p, int q) {
        //Keep on substracting q from p and update directions
        boolean up = true;
        boolean east = true;
        int remain = p;
        while (true) {
            remain -= q;
            if (remain == 0) {
                if(!east) return 2;
                return up ? 1 : 0;
            }
            if (remain < 0) {
                remain+=p;
                up=!up;
            }
            east = !east;
        }
    }
}
